src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java

```diff
@@ -0,0 +1,368 @@
+import com.hivemq.cli.commands.options.SslOptions;
+import com.hivemq.cli.converters.MqttVersionConverter;
+import com.hivemq.cli.mqtt.test.Mqtt3FeatureTester;
+import com.hivemq.cli.mqtt.test.Mqtt5FeatureTester;
+import com.hivemq.cli.mqtt.test.results.*;
+import com.hivemq.client.mqtt.MqttClientSslConfig;
+import com.hivemq.client.mqtt.MqttVersion;
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt3.message.connect.connack.Mqtt3ConnAck;
+import com.hivemq.client.mqtt.mqtt3.message.connect.connack.Mqtt3ConnAckReturnCode;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAck;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAckReasonCode;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAckRestrictions;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.tinylog.Logger;
+import org.tinylog.configuration.Configuration;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@CommandLine.Command(
+        name = "test",
+        description = "Tests the specified broker on different MQTT feature support and prints the results")
+public class TestBrokerCommand implements Runnable {
+
+    final int MAX_PAYLOAD_TEST_SIZE = 100000; // ~ 1 MB
+
+    @CommandLine.Option(names = {"-h", "--host"}, description = "The hostname of the message broker (default 'localhost')", order = 1)
+    private @Nullable String host;
+
+    @CommandLine.Option(names = {"--help"}, usageHelp = true, description = "display this help message")
+    boolean usageHelpRequested;
+
+    @CommandLine.Option(names = {"-p", "--port"}, description = "The port of the message broker (default: 1883)", order = 1)
+    private @Nullable Integer port;
+
+    @CommandLine.Option(names = {"-V", "--mqttVersion"}, converter = MqttVersionConverter.class, description = "The mqtt version to test the broker on", order = 1)
+    private @Nullable MqttVersion version;
+
+    @CommandLine.Option(names = {"-f", "--force"}, defaultValue = "false", description = "Force test MQTT 5 features (even if MQTT3 features were already tested)", order = 1)
+    private boolean force;
+
+    @CommandLine.Option(names = {"-t", "--timeOut"}, defaultValue = "10", description = "The time to wait for the broker to respond", order = 1)
+    private @NotNull Integer timeOut;
+
+    @CommandLine.Option(names = {"-q", "--qosTries"}, defaultValue = "10", description = "The amount of publishes to send to the broker on every qos level", order = 1)
+    private @NotNull Integer qosTries;
+
+    @CommandLine.Mixin
+    private AuthenticationOptions authenticationOptions = new AuthenticationOptions();
+
+    @CommandLine.Mixin
+    private SslOptions sslOptions = new SslOptions();
+
+    private @Nullable MqttClientSslConfig sslConfig;
+
+    //needed for pico cli - reflection code generation
```

Comment:
this comment probably belongs to the constructor?

Reply:
Yes.